middleware/file/lookup.go

```diff
@@ -78,6 +79,15 @@ func (z *Zone) Lookup(state request.Request, qname string) ([]dns.RR, []dns.RR,
 
 		elem, found = z.Tree.Search(parts)
 		if !found {
+			// Look to see whether the last label matched has a DNAME record.
```

Comment:
This looks pretty expensive. Maybe we should do this, if we know there aren't any DNAMEs in the zone?